From 96487c8ea92700962d6449533627f348cfc53632 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 21 Oct 2016 04:40:23 +0100
Subject: [PATCH] systemctl: Don't emit a reload warning for systemd-cryptsetup

This is modified by dracut during early boot, so instead of a nice clean
prompt for the password, the user is presented with a scary warning about
things being modified on disk, that they cannot possibly understand isn't
actually a bad thing.

This simply skips all reload warnings for the transient dracut unit on
Solus for systemd-cryptsetup, as this was generated *only* for the boot
unit.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/systemctl/systemctl.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/systemctl/systemctl.c b/src/systemctl/systemctl.c
index b1441ad..b7659c8 100644
--- a/src/systemctl/systemctl.c
+++ b/src/systemctl/systemctl.c
@@ -2663,10 +2663,16 @@ static int start_unit_one(
         if (r < 0)
                 return bus_log_parse_error(r);
 
-        if (need_daemon_reload(bus, name) > 0)
+        if (need_daemon_reload(bus, name) > 0) {
+                /* Skip systemd-cryptsetup@ in early boot. No need for fear. */
+                if (strncmp(name, "systemd-cryptsetup@", 19) == 0 && arg_scope == UNIT_FILE_SYSTEM) {
+                        goto skip_warning;
+                }
                 log_warning("Warning: Unit file of %s changed on disk, 'systemctl%s daemon-reload' recommended.",
                             name, arg_scope == UNIT_FILE_SYSTEM ? "" : " --user");
+        }
 
+skip_warning:
         if (s) {
                 char *p;
 
-- 
2.11.0

